### PR TITLE
CI builds for Linux

### PR DESCRIPTION
**Pull request by slipher**
Linux native builds, one with GCC and one with Clang, just like the
Travis configuration which stopped working a few months ago.

It said the Ubuntu 16.04 environment will be removed on October 18, 2021
so I used 18.04 (Travis was 16.04 IIRC).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+    - bash: |
+        set -e
+        sudo apt-get update
+        sudo apt-get -y -q --no-install-recommends install zlib1g-dev libncursesw5-dev libgeoip-dev nettle-dev libgmp-dev libcurl4-gnutls-dev libsdl2-dev libogg-dev libvorbis-dev libopusfile-dev libtheora-dev libwebp-dev libjpeg8-dev libpng-dev libfreetype6-dev libglew-dev libopenal-dev liblua5.2-dev ninja-build
+      displayName: 'Install deps'
+    - bash: |
+        set -e
+        git submodule update --init --recursive
+        cmake --version
+        cmake -G "Ninja" -Wdev -Wdeprecated -DCMAKE_C_COMPILER=$(C_COMPILER) -DCMAKE_CXX_COMPILER=$(CXX_COMPILER) -DUSE_PRECOMPILED_HEADER=0 -DUSE_WERROR=1 -DBE_VERBOSE=1 -DUSE_DEBUG_OPTIMIZE=0 -DCMAKE_BUILD_TYPE=Debug -H. -Bbuild
```
Comment (illwieckz) [inline azure-pipelines.yml:37]: This is bikesheding, but the native CMake boolean values are `ON` and `OFF`, CMake automatically converts `0`, `1`, `true` and `false` to either `ON` or `OFF`.

Comment (slipher) [reply azure-pipelines.yml:37]: Well, 0/1 is consistent with the other files in the repo (other CI and external_deps/build.sh).

Comment (illwieckz) [reply azure-pipelines.yml:37]: OK, let's favor consistency for now then. We can do restyle later.

**Review by illwieckz — APPROVED** [review]
LGTM.

**Pull request merged**